### ISSUE TITLE
Abuse of catch(Throwable)

### ISSUE THREAD

**Issue by dktapps** [pmmp/PocketMine-MP#2072]
### Issue description
Since PHP 7, many things that were previously fatal errors in PHP 5 are now catchable thrown `Error` objects. The `Throwable` interface is implemented by both `Error` and `Exception` to allow what should be a **rare** catch-all-things case, for example for logging or crashdumps.

PocketMine-MP heavily abuses `catch(\Throwable)` since PHP 7, which results in many things that ought to cause crashdumps, to produce logged exceptions and then attempt to continue. This is bad behaviour. `catch(\Throwable)` should only be used in a handful of places - most other things should only be catching `Exception`s. `Error`s are usually thrown to indicate problems with user code, _not_ runtime errors.

As seen [here](https://github.com/tpunt/PHP7-Reference#exceptions-in-the-engine), `Error`s were explicitly segregated from `Exception`s to _maintain backwards compatibility_ to some degree, and **prevent** existing catches catching `Error`s. Catch-all everywhere is a bad thing.

### Steps to reproduce the issue
1. trigger an `Error` somewhere down the line
2. behold the attempt to continue execution wrongfully on a case that should cause a crashdump to be emitted.

### OS and versions
* PocketMine-MP: any since PHP 7
* PHP: 7.any

**Comment by dktapps** [comment]
A quick example:

```php
//assume we are loading a chunk
try{
    $chunk = $this->provider->loadChunk($chunkX, $chunkZ); //we forgot to set chunkZ to a number
}catch(\Throwable $e){
    $this->server->getLogger()->error("Corrupted chunk detected: " . $e->getMessage()); //this isn't a corrupted chunk, it's a TypeError due to defective code!
}
```

**Comment by dktapps** [comment]
This has been fully addressed on master as of 48a99937b9556d0b04121d58ea45dce994d88633.
